app/login/login.js

```diff
@@ -73,9 +73,15 @@ class Login extends Component {
           sameSite: 'strict',
         });
         cookie.set('CDAP_Auth_User', this.state.username, { secure: isSecure, sameSite: 'strict' });
-        var queryObj = util.getQueryParams(location.search);
-        queryObj.redirectUrl = queryObj.redirectUrl || '/';
-        window.location.href = queryObj.redirectUrl;
+        const queryObj = util.getQueryParams(location.search);
+        let validRedirectUrl;
+        // Require that redirectUrl be a relative path to avoid open redirects
+        if (queryObj.redirectUrl && queryObj.redirectUrl.startsWith('/')) {
```

Comment:
This works for the case in which the redirect is on the same domain, but might not work for the case in which the the redirect is for a different domain (e.g. login server and CDAP are hosted on different domains). We should double check to see if this is something we want to continue supporting

Reply:
Offline synced, we don't support redirection from a different domain, should be fine for the fix

Reply:
SGTM, thank you!